plugins/chat/assets/javascripts/discourse/components/chat-channel.js

```diff
@@ -704,7 +701,7 @@ export default class ChatLivePane extends Component {
           return;
         }
 
-        this.args.channel.draft = null;
+        this.chatDraftsManager.remove({ channelId: this.args.channel.id });
```

Comment:
Both these changes ensure we have removed this draft as soon as the message is sent.